Sources/MongoDriver.swift

```
@@ -2,176 +2,267 @@ import Foundation
+                throw Error.invalidData
+            }
+            
+            document[key] = value
         }
         
-        if let limit = query.limit {
-            cursor = try database[query.entity].find(matching: mkq,
-                                                     sortedBy: sortDocument,
-                                                     skipping: Int32(limit.offset),
-                                                     limitedTo: Int32(limit.count))
-        } else {
-            cursor = try database[query.entity].find(matching: mkq,
-                                                     sortedBy: sortDocument)
+        if let key = document.type(at: "_id"), case ElementType.nullValue = key {
+            document["_id"] = nil
         }
-
-        return cursor
+        
+        return document
     }
-
-    private func modify<T: Entity>(_ query: Fluent.Query<T>) throws {
-        guard let data = query.data?.nodeObject else {
-            throw Error.noData
+    
+    /// Exequtes a Fluent.Query for an Entity
+    ///
+    ///
+    public func query<E>(_ query: RawOr<Fluent.Query<E>>) throws -> Node where E : Entity {
+        guard let query = query.wrapped else {
+            throw Error.invalidQuery
         }
-
-
-        let aqt = try query.makeAQT()
-        let mkq = MKQuery(aqt: aqt)
-
-        var document: Document = [:]
-
-        for (key, val) in data {
-            if key == idKey {
-                continue
+        
+        let collection = self[E.entity]
+        let filter = try makeQuery(query.filters, method: .and)
+        let document = try makeDocument(query.data)
+        let sort = makeSort(query.sorts)
+        let (limit, skip) = try makeLimits(query.limits)
+        
+        switch query.action {
+        case .create:
+            return try collection.insert(document).makeNode()
+        case .fetch(_):
+            // TODO: Support ComputedProperties
```

We should throw `unsupported` error if the computed properties array is not empty